From b091aac5b60202c5258fb07094858ffdc5593d1f Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Mon, 12 Oct 2026 13:50:58 +0100
Subject: [PATCH 15/16] always honour the global exec spawning default

drop the per-app compat-config override for secure app spawning: apps
in the appcompatconfig table (notably gms) were forced onto the exec
spawning path, which re-execs app_process and loses all zygote hook
injection (zygisk). every app now follows the global
persist.security.exec_spawn default.
---
 core/java/android/ext/settings/app/AswUseExecSpawning.java | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/core/java/android/ext/settings/app/AswUseExecSpawning.java b/core/java/android/ext/settings/app/AswUseExecSpawning.java
index a833aae2c..b4058f844 100644
--- a/core/java/android/ext/settings/app/AswUseExecSpawning.java
+++ b/core/java/android/ext/settings/app/AswUseExecSpawning.java
@@ -7,8 +7,6 @@ import android.content.pm.GosPackageState;
 import android.content.pm.GosPackageStateFlag;
 import android.ext.settings.ExtSettings;
 
-import com.android.server.os.nano.AppCompatProtos;
-
 /** @hide */
 public class AswUseExecSpawning extends AppSwitch {
     public static final AswUseExecSpawning I = new AswUseExecSpawning();
@@ -16,7 +14,6 @@ public class AswUseExecSpawning extends AppSwitch {
     private AswUseExecSpawning() {
         gosPsFlag = GosPackageStateFlag.USE_EXEC_SPAWNING;
         gosPsFlagNonDefault = GosPackageStateFlag.USE_EXEC_SPAWNING_NON_DEFAULT;
-        compatChangeToDisableHardening = AppCompatProtos.USE_ZYGOTE_SPAWNING;
     }
 
     public static boolean isEnabledFor(Context ctx, @UserIdInt int userId, ApplicationInfo appInfo,
-- 
2.55.0

